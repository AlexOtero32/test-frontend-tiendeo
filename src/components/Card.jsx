import React from 'react';

const Card = ({ titulo, descripcion, url }) => (
    <div className="card">
        <img
            className="w-full"
            src={url || 'https://placehold.it/400x200'}
            alt={titulo}
        />
        <div className="px-6 py-4 w-full">
            <div className="font-bold text-xl mb-2 text-gray-600">{titulo}</div>
            <p className="text-gray-700 text-base">{descripcion}</p>
        </div>
        <div className="px-6 py-4">
            <span className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2">
                #photography
            </span>
            <span className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2">
                #travel
            </span>
            <span className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700">
                #winter
            </span>
        </div>
    </div>
);

export default Card;
