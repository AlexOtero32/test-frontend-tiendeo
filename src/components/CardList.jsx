import React from 'react';
import { connect } from 'react-redux';

import Card from './Card';

const CardList = (props) => {
    return (
        <main className="container grid-container">
            {props.cards.map((card) => {
                return (
                    <Card
                        key={card.id}
                        id={card.id}
                        titulo={card.titulo}
                        descripcion={card.descripcion}
                        url={card.url}
                    />
                );
            })}
        </main>
    );
};

const mapStateToProps = (state) => ({
    cards: state.cards.cardList,
});

export default connect(mapStateToProps, null)(CardList);
