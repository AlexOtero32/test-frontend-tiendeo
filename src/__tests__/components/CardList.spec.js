import React from "react";
import Card from "../../components/Card";
import CardList from "../../components/CardList";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureStore([]);

describe("CardList component", function () {
  let store;

  const createCard = () => ({
    date: new Date(),
    titulo: "Tarjeta",
    id: Math.random().toString(),
    descripcion: "DescripciÃ³n",
    url: undefined,
  });

  beforeEach(() => {
    store = mockStore({
      cards: {
        cardList: [createCard(), createCard(), createCard()],
      },
    });
  });

  it("Renderiza correctamente", function () {
    const component = renderer.create(
      <Provider store={store}>
        <CardList />
      </Provider>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Renderiza las Cards correctamente", function () {
    const component = renderer.create(
      <Provider store={store}>
        <CardList />
      </Provider>
    );

    const wrapper = component.root.findByType("main");

    expect(wrapper.children.length).toBe(3);
  });
});
