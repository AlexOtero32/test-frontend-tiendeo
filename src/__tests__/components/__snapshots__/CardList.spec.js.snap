// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component Renderiza correctamente 1`] = `
<main
  className="container grid-container"
>
  <div
    className="card"
  >
    <div
      className="image-box"
    >
      <img
        alt="Tarjeta"
        className="w-full"
        src="https://placehold.it/400x200"
      />
      <div
        className="button-box"
      >
        <button
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-xl font-semibold text-gray-700 mr-2"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </button>
        <button
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-xl font-semibold text-gray-700 mr-2"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="px-6 py-4 w-full"
    >
      <div
        className="font-bold text-xl mb-2 text-gray-600"
      >
        Tarjeta
      </div>
      <p
        className="text-gray-700 text-base"
      >
        Descripción
      </p>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="image-box"
    >
      <img
        alt="Tarjeta"
        className="w-full"
        src="https://placehold.it/400x200"
      />
      <div
        className="button-box"
      >
        <button
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-xl font-semibold text-gray-700 mr-2"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </button>
        <button
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-xl font-semibold text-gray-700 mr-2"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="px-6 py-4 w-full"
    >
      <div
        className="font-bold text-xl mb-2 text-gray-600"
      >
        Tarjeta
      </div>
      <p
        className="text-gray-700 text-base"
      >
        Descripción
      </p>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="image-box"
    >
      <img
        alt="Tarjeta"
        className="w-full"
        src="https://placehold.it/400x200"
      />
      <div
        className="button-box"
      >
        <button
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-xl font-semibold text-gray-700 mr-2"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </button>
        <button
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-xl font-semibold text-gray-700 mr-2"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="px-6 py-4 w-full"
    >
      <div
        className="font-bold text-xl mb-2 text-gray-600"
      >
        Tarjeta
      </div>
      <p
        className="text-gray-700 text-base"
      >
        Descripción
      </p>
    </div>
  </div>
</main>
`;
